core/templates/dev/head/exploration_editor/main_tab/StateResponses.js

```diff
@@ -306,6 +315,58 @@ oppia.controller('StateResponses', [
         return choiceIndices.every(function(choiceIndex) {
           return handledAnswersArray.indexOf(choiceIndex) !== -1;
         });
+      } else if (interactionId === 'ItemSelectionInput') {
+        var maxSelectionCount = (
+            customizationArgs.maxAllowableSelectionCount.value);
+        if (maxSelectionCount === 1) {
+          var numChoices = $scope.getAnswerChoices().length;
+          // Use array to store booleans that become true
+          // if answerChoice is covered
+          handledAnswersArray = [];
+          for (var i = 0; i < numChoices; i++) {
+            handledAnswersArray.push(false);
+            answerChoices.push($scope.getAnswerChoices()[i].val);
+          }
+
+          var newHandledAnswers = function(current) {
+            return handledAnswersArray.map(function(element, index) {
+              if (index !== current || element === true) {
+                return true;
+              } else if (index === current && element !== true) {
+                return false;
+              }
+            });
+          };
+
+          answerChoices.forEach(function(answerChoice, choiceIndex) {
```

Comment:
From a correctness perspective, this looks good! It's a bit inefficient, because you're iterating through all answer groups for every single answer choice. But it does the job and I think it's pretty clear.

For reference, here's another approach to doing something like this which gets rid of the outer loop (thus speeding things up by a factor of answerChoices.length):

```
// First create a JS object called answerChoiceToIndex that maps answer choice strings to their indexes. Then:

answerGroups.forEach(function(answerGroup) {
  answerGroup.rule_specs.forEach(function(ruleSpec) {
    var ruleInputs = ruleSpec.inputs.x;
    ruleInputs.forEach(function(ruleInput) {
      var choiceIndex = answerChoiceToIndex[ruleInput];  // this is where you use the JS object mentioned above
      if (ruleSpec.rule_type === 'Equals' || ruleSpec.rule_type === 'ContainsAtLeastOneOf') {
        handledAnswersArray[choiceIndex] = true;
      } else if (ruleSpec.rule_type === 'DoesNotContainAtLeastOneOf') {
        for (var i = 0; i < handledAnswersArray.length; i++) {
          if (handledAnswersArray[i] !== choiceIndex) {
            handledAnswersArray[choiceIndex] = true;
          }
        }
      }
    }
  }
}
```

I'm fine either way though. JS is quite fast and we aren't going to have thousands/millions of answer choices, so either algorithm is fine IMO.